models/vrp.rb

```diff
@@ -337,49 +337,96 @@ def self.remove_unecessary_units(hash)
       hash[:shipments]&.map{ |s| (s[:quantities] || []).delete_if{ |quantity| needed_units.exclude? quantity[:unit_id] } }
     end
 
-    def self.generate_schedule_indices_from_date(hash)
-      return hash if !hash[:configuration] || !hash[:configuration][:schedule] ||
-                     hash[:configuration][:schedule][:range_indices]
+    def self.convert_availability_dates_into_indices(element, hash, start_index, type)
+      unavailable_indices, unavailable_dates =
+        case type
+        when :visit
+          [:unavailable_visit_day_indices, :unavailable_visit_day_date]
```

Comment:
Why don't we make the rest of the code uniform like the way we did for the API.

That is, we have `unavailable_index_ranges` at the API level for services/shipments/vehicle/config and we can do `unavailable_indices` for all inside the code base instead of `unavailable_visit_day_indices`/`unavailable_work_day_indices`/`unavailable_indices` and `unavailable_visit_day_date`/`unavailable_visit_day_date`/`unavailable_date`. 

This separation (of the same exact logic "unavailability") makes the code more complex for no gain. For example here we needed to jump through hoops to make the parsing work for all objects.

Reply:
Do you mean making names uniform within the code ? (but still having _unavailable_indices_ for each vehicle/service/shipment/schedule ?). We could do that in our internal model, you are right.